gui/src/components/widgets/SkeletonVisualizerWidget.tsx

```diff
@@ -1,122 +1,132 @@
-import { Canvas } from '@react-three/fiber';
+import { Canvas, Object3DNode, extend, useThree } from '@react-three/fiber';
 import { useAppContext } from '@/hooks/app';
-import { Bone, PerspectiveCamera, Quaternion, Skeleton } from 'three';
-import { getHelperFromSkeleton } from 'three/examples/jsm/utils/SkeletonUtils';
-import { useMemo } from 'react';
-import { Vector3FromVec3fT } from '@/maths/vector3';
-import { QuaternionFromQuatT } from '@/maths/quaternion';
+import { Bone } from 'three';
+import { useMemo, useEffect, useRef } from 'react';
+import { OrbitControls, OrthographicCamera } from '@react-three/drei';
+import {
+  BoneKind,
+  createChildren,
+  BasedSkeletonHelper,
+} from '@/utils/skeletonHelper';
+import * as THREE from 'three';
+import { BodyPart, BoneT } from 'solarxr-protocol';
 
-export const threeSkeleton = (() => {
-  const bones = [];
+extend({ BasedSkeletonHelper });
 
-  const head = new Bone();
-  const neck = new Bone();
-  head.add(neck);
+declare module '@react-three/fiber' {
+  interface ThreeElements {
+    basedSkeletonHelper: Object3DNode<
+      BasedSkeletonHelper,
+      typeof BasedSkeletonHelper
+    >;
+  }
+}
 
-  const chest = new Bone();
-  neck.add(chest);
-  const waist = new Bone();
-  chest.add(waist);
-  const hip = new Bone();
-  waist.add(hip);
+const groundColor = '#4444aa';
+const frustumSize = 10;
+const factor = 2;
 
-  const leftShoulder = new Bone();
-  neck.add(leftShoulder);
-  const leftUpperArm = new Bone();
-  leftShoulder.add(leftUpperArm);
-  const leftLowerArm = new Bone();
-  leftUpperArm.add(leftLowerArm);
-  const leftHand = new Bone();
-  leftLowerArm.add(leftHand);
+export function OrthographicCameraWrapper() {
+  const { size } = useThree();
+  const aspect = useMemo(() => size.width / size.height, [size]);
 
-  const rightShoulder = new Bone();
-  neck.add(rightShoulder);
-  const rightUpperArm = new Bone();
-  rightShoulder.add(rightUpperArm);
-  const rightLowerArm = new Bone();
-  rightUpperArm.add(rightLowerArm);
-  const rightHand = new Bone();
-  rightLowerArm.add(rightHand);
+  return (
+    <OrthographicCamera
+      makeDefault
+      zoom={200}
+      top={frustumSize / factor}
+      bottom={frustumSize / -factor}
+      left={(0.5 * frustumSize * aspect) / -factor}
+      right={(0.5 * frustumSize * aspect) / factor}
+      near={0.1}
+      far={1000}
+      position={[25, 75, 50]}
+    />
+  );
+}
 
-  const leftUpperLeg = new Bone();
-  hip.add(leftUpperLeg);
-  const leftLowerLeg = new Bone();
-  leftUpperLeg.add(leftLowerLeg);
-  const leftFoot = new Bone();
-  leftLowerLeg.add(leftFoot);
+const yParts = [
+  BodyPart.NECK,
+  BodyPart.UPPER_CHEST,
+  BodyPart.CHEST,
+  BodyPart.WAIST,
+  BodyPart.HIP,
+  BodyPart.LEFT_UPPER_LEG,
+  BodyPart.LEFT_LOWER_LEG,
+];
```

Comment:
Might be worth a comment explaining why right leg is missing. I'm assuming this is just to get the total height?

Reply:
yep